### PR TITLE
CODENVY-1757; added RedHat OAuth authentication

### LINKED ISSUE #1757: Red Hat OAuth2 authentication provider
Implement Red Hat Oauth2 authentication provider

```
I found this URL which returns descriptor of the OpenID Connect/OAuth endpoint: https://developers.redhat.com/auth/realms/rhd/.well-known/openid-configuration

I believe "userinfo_endpoint" field contains url your colleague asked me for. 
```

### PR DESCRIPTION
**Pull request by mshaposhnik**
### What does this PR do?
Added OAuth2 authentication via RedHat developers network.

Enabling/disabling this provider by setting `CODENVY_REDHAT_CLIENT_ID` and `CODENVY_REDHAT_SECRET` in codenvy.env

### What issues does this PR fix or reference?
Fixes #1757 

#### Changelog
Added oAuth2 authentication with the Red Hat Developer Network.

#### Release Notes
n/a

#### Docs PR
n/a

### CODE REVIEW

File: dockerfiles/init/modules/codenvy/templates/oauth.properties.erb [hunk 1]
```diff
@@ -47,3 +47,11 @@ oauth.microsoft.authuri=https://app.vssps.visualstudio.com/oauth2/authorize
...
 oauth.microsoft.useruri=https://app.vssps.visualstudio.com/_apis/profile/profiles/me?api-version=1.0
 oauth.microsoft.git.pattern=https://([0-9a-zA-Z-_.%]+)\\.visualstudio\\.com/.*/?_git/.+
+
+
+oauth.redhat.clientid=<%= scope.lookupvar('codenvy::redhat_client_id') %>
+oauth.redhat.clientsecret=<%= scope.lookupvar('codenvy::redhat_secret') %>
+oauth.redhat.redirecturis=<%= scope.lookupvar('codenvy::host_protocol') %>://<%= scope.lookupvar('codenvy::host_url') %>/api/oauth/callback
+oauth.redhat.authuri=https://developers.redhat.com/auth/realms/rhd/protocol/openid-connect/auth
+oauth.redhat.tokenuri=https://developers.redhat.com/auth/realms/rhd/protocol/openid-connect/token
+oauth.redhat.useruri=https://developers.redhat.com/auth/realms/rhd/protocol/openid-connect/userinfo
```
Comment (riuvshin) [inline dockerfiles/init/modules/codenvy/templates/oauth.properties.erb]: please add new line

**Comment by bmicklea** [comment]
@mshaposhnik - this is going to be part of the default codenvy.io login options?

**Comment by mshaposhnik** [comment]
@bmicklea I don't know the right answer. I just provided ability. Ask @TylerJewell for 100% guarantee.

**Comment by bmicklea** [comment]
Then let's add it for SaaS on Codenvy.io by default - more options for painless account creation is better.

**Comment by skabashnyuk** [comment]
@bmicklea  we can decided it at any time. By default it's not configured

**Pull request merged**